src/components/MenuItem/MenuItem.tsx

```diff
@@ -0,0 +1,92 @@
+/*
+Copyright 2023 New Vector Ltd
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import classnames from "classnames";
+import React, {
+  ComponentPropsWithoutRef,
+  ComponentType,
+  ElementType,
+  SVGAttributes,
+} from "react";
+import styles from "./MenuItem.module.css";
+import { Text } from "../Typography/Text";
+import ChevronRightIcon from "@vector-im/compound-design-tokens/icons/chevron-right.svg";
+
+type MenuItemElement = "button" | "label" | "div";
+
+type Props<C extends MenuItemElement> = {
+  /**
+   * The element type of this menu item.
+   * @default button
+   */
+  as?: C;
+  className?: string;
+  /**
+   * The icon to show on this menu item.
+   */
+  Icon: ComponentType<SVGAttributes<SVGElement>>;
+  /**
+   * The label to show on this menu item.
+   */
+  label: string;
+  /**
+   * The color variant of the menu item.
+   * @default primary
+   */
+  kind?: "primary" | "critical";
+  /**
+   * Whether to replace the children with a navigation hint on hover.
```

Comment:
This comment does not look accurate compared to the implementation.
It seems to add a chevron rather than it replacing the children?

Reply:
The chevron needs to be displayed all the time. it's a mistake on the Figma design that it is missing.
We can therefore remove that prop

Reply:
I guess we figured this out in [in discussion](https://github.com/vector-im/compound-web/pull/91#discussion_r1338556370)

Reply:
oh, but: the replacement of the children happens in CSS, and I can see why that might not be obvious. I've added a couple comments to hopefully make that clearer.